Core/COObject.m

```diff
@@ -1305,6 +1323,9 @@ - (void)validateNewValue: (id)newValue
 	ETModelDescriptionRepository *repo = _objectGraphContext.modelDescriptionRepository;
 	BOOL isPersistentRelationship = propertyDesc.isPersistentRelationship;
 	BOOL isValidatableType = ([self serializationGetterForProperty: propertyDesc.name] == NULL);
+	
+	if (!isValidatableType)
+		return;
```

Comment:
It's more a temporary workaround until serialization accessors get removed. This hack is already present in the slow path.

This could hide bugs but basically serialization accessors can return arbitrary objects, and this validation code would need to be changed to check ETPropertyDescription.persistentType if we want to run all these assertions. Since the code is going away soon, I don't think it's worth the investment.